src/backend/distributed/sql/udfs/get_missing_time_partition_ranges/latest.sql

```diff
@@ -0,0 +1,264 @@
+
+        IF table_partition_interval < INTERVAL '1 day' THEN
+            /* include time of day in partition name */
+            datetime_string_format := datetime_string_format || '_HH24MI';
+        END IF;
+
+        IF table_partition_interval < INTERVAL '1 minute' THEN
+             /* include seconds in time of day in partition name */
+             datetime_string_format := datetime_string_format || 'SS';
+        END IF;
+    END IF;
+
+    WHILE current_range_from_value < to_date LOOP
+        /*
+         * Check whether partition with given range has already been created
+         * Since partition interval can be given with different types, we are converting
+         * all variables to timestamptz to make sure that we are comparing same type of parameters
+         */
+        PERFORM * FROM pg_catalog.time_partitions
+        WHERE
+            from_value::timestamptz = current_range_from_value::timestamptz AND
+            to_value::timestamptz = current_range_to_value::timestamptz AND
+            parent_table = table_name;
+        IF found THEN
+            current_range_from_value := current_range_to_value;
+            current_range_to_value := current_range_to_value + table_partition_interval;
+            CONTINUE;
+        END IF;
+
+        /*
+         * Check whether any other partition covers from_value or to_value
+         * That means some partitions doesn't align with the initial partition.
+         * In other words, gap(s) exist between partitions which is not multiple of intervals.
+         */
+        SELECT from_value::text, to_value::text
+        INTO manual_partition_from_value_text, manual_partition_to_value_text
+        FROM pg_catalog.time_partitions
+        WHERE
+            ((current_range_from_value::timestamptz > from_value::timestamptz AND current_range_from_value < to_value::timestamptz) OR
+            (current_range_to_value::timestamptz > from_value::timestamptz AND current_range_to_value::timestamptz < to_value::timestamptz)) AND
+            parent_table = table_name;
+
+        IF found THEN
+            RAISE 'Partition with the range from % to % does not align with the initial partition. Please make sure that no gap(s) exists between existing partitions', -- TODO: Check the message
+            manual_partition_from_value_text,
+            manual_partition_to_value_text;
+        END IF;
+
+        IF table_partition_column_type_name = 'date' THEN
+            SELECT current_range_from_value::date::text INTO current_range_from_value_text;
+            SELECT current_range_to_value::date::text INTO current_range_to_value_text;
+        ELSIF table_partition_column_type_name = 'timestamp without time zone' THEN
+            SELECT current_range_from_value::timestamp::text INTO current_range_from_value_text;
+            SELECT current_range_to_value::timestamp::text INTO current_range_to_value_text;
+        ELSIF table_partition_column_type_name = 'timestamp with time zone' THEN
```

Comment:
this check feels a bit more reliable if we were using table_partition_column_type_id = timestamptz::regtype 

(not sure if "timestampt with time zone" is subject to internationalization)